packages/sync-actions/src/zones-actions.js

```diff
@@ -19,16 +16,40 @@ export function actionsMapBase(diff, oldObj, newObj) {
 }
 
 export function actionsMapLocations(diff, oldObj, newObj) {
-  const handler = createBuildArrayActions('locations', {
-    [ADD_ACTIONS]: newLocation => ({
-      action: 'addLocation',
-      location: newLocation,
-    }),
-    [REMOVE_ACTIONS]: removedLocation => ({
-      action: 'removeLocation',
-      location: removedLocation,
-    }),
+  const actions = []
+  if (!diff.locations) return actions
+
+  const deleteActions = []
+  const addActions = []
+
+  forEach(diff.locations, (location, index) => {
```

Comment:
Why did you move away from `createBuildArrayActions`?

Reply:
For what stated above in the description of the PR, I need to handle the locations changes "manually" to taking into account the location "swapping" (I don't know how to call it honestly 😅)

Reply:
I see. Lets wait for @wizzy25 feedback on this then.

Reply:
Absolutely, I would love to know if there's a better approach for that case or not. This really gave me some headaches this last days haha

Reply:
@qmateub Instead of this, why not consider using the `createBuildArrayActions`, and modifying your code as such:
```js
// zone-actions.js
export function actionsMapLocations(diff, oldObj, newObj) {
  const handler = createBuildArrayActions('locations', {
    [ADD_ACTIONS]: newLocation => ({
      action: 'addLocation',
      location: newLocation,
    }),
    [REMOVE_ACTIONS]: oldLocation => ({
      action: 'removeLocation',
      location: oldLocation,
    }),
    [CHANGE_ACTIONS]: (oldLocation, newLocation) => [
      {
        action: 'removeLocation',
        location: oldLocation,
      },
      {
        action: 'addLocation',
        location: newLocation,
      },
    ],
  })

  return handler(diff, oldObj, newObj)
}
```
Then add one more flatten in your `zones` file
```js
// zones.js
function createZonesMapActions(mapActionGroup) {
  return function doMapActions(diff, newObj, oldObj /* , options */) {
    const allActions = []

    allActions.push(
      mapActionGroup('base', () =>
        zonesActions.actionsMapBase(diff, oldObj, newObj)
      )
    )

    allActions.push(
      flatten( // <- new flatten added here because we returned 2 objects in actions file
        mapActionGroup('locations', () =>
          zonesActions.actionsMapLocations(diff, oldObj, newObj)
        )
      )
    )

    return flatten(allActions)
  }
}
```